### ISSUE TITLE
async_log_helper.h(220):  C4355: 'this': used in base member initializer list

### ISSUE THREAD

**Issue by benstieg** [gabime/spdlog#637]
Hi, 

according to [docs.microsoft.com](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-c4355) the constructor of [`async_log_helper`](https://github.com/gabime/spdlog/blob/v0.16.3/include/spdlog/details/async_log_helper.h#L220) 
```
    _worker_teardown_cb(worker_teardown_cb),
    _worker_thread(&async_log_helper::worker_loop, this)
{}
```

should not be more like
```
 _worker_teardown_cb(worker_teardown_cb)
{
  _worker_thread = std::thread(&async_log_helper::worker_loop, this);
}
```

**Comment by gabime** [comment]
The docs say that base classes should not be initialized with ```this```.. ```_worker_thread``` is not base class

**Comment by benstieg** [comment]
We are using pretty strickt compiler warning settings in our projects and when we stared using spdlog with Visual Studio 2017 we got the C4355 warning for async_log_helper.h(220). 
I can see that in this particular case it will probably not cause any problem, but in general I believe that compiler warnings should rather be fixed than ignored. 

PS: this warning was the only one which we got from the spdlog headers which implies a very good code quality and spdlog works great.
